packages/context/test/unit/resolver.test.ts

```diff
@@ -320,3 +326,90 @@ function customAsyncDecorator(def: Object) {
     return f + ':' + b;
   });
 }
+
+describe('method injection', () => {
+  let ctx: Context;
+
+  before(function() {
+    ctx = new Context();
+    ctx.bind('foo').to('FOO');
+    ctx.bind('bar').to('BAR');
+  });
+
+  it('can resolve method arguments', () => {
+    class TestClass {
+      test(@inject('foo') foo: string) {
+        return `hello ${foo}`;
+      }
+    }
+
+    const t = invokeMethod(TestClass.prototype, 'test', ctx);
```

Comment:
I find this hacky, is it an expected usage to invoke prototype methods directly on `TestClass.prototype`, thus passing a wrong `this` value?

Reply:
(Same comment probably applies to "async" tests below too.)